charts/stable/otel-collector/templates/secret.yaml

```diff
@@ -0,0 +1,18 @@
+{{/*
+The open telemetry config secret to be included.
+*/}}
+{{- if and .Values.configFile (not .Values.configFileSecret) }}
+---
+apiVersion: v1
+kind: Secret
```

Comment:
What is the reason for creating this as a Secret? I can see that there are potentially API keys being stored in here, but since they would be wide-open in the values.yaml already, there seems to be no real need for a Secret?

Reply:
I think assuming it's always a secret makes it easier to include the user-supplied one or the helm-configured one. In some situations it may not have any sensitive information in it and could be a configmap, but since that would require a lot of logic to support and doesn't benefit the user, I just leave it as a secret here. 

The admin can decide of they want the API key in a helm value yaml or if they want to handle that outside the chart, make a secret manually, and then reference it.